be/src/olap/rowset/segment_v2/segment_writer.cpp

```diff
@@ -587,6 +598,17 @@ Status SegmentWriter::append_block(const vectorized::Block* block, size_t row_po
                         << "found duplicate key or key is not sorted! current key: " << key
                         << ", last key" << last_key;
                 RETURN_IF_ERROR(_primary_key_index_builder->add_item(key));
+                if (is_unique_key_ignore_mode) {
+                    RowLocation loc;
+                    RowsetSharedPtr rowset;
+                    auto status = _tablet->lookup_row_key(key, false, specified_rowsets, &loc,
```

Comment:
Don't check if key exists here, you should change the behavior of all the places of update delete bitmap